src/main/java/iudx/resource/server/database/async/AsyncServiceImpl.java

```diff
@@ -90,30 +102,57 @@ public AsyncService asyncStatus(
             if (results.isEmpty()) {
               responseBuilder =
                   new ResponseBuilder("failed")
-                      .setTypeAndTitle(400)
+                      .setTypeAndTitle(400, BAD_REQUEST.getUrn())
                       .setMessage("Fail: Incorrect search ID");
               handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
             } else {
               JsonObject answer = results.getJsonObject(0);
+              LOGGER.debug("answer {}", answer);
 
               String userId = answer.getString("user_id");
               if (sub.equals(userId)) {
                 String status = answer.getString(STATUS);
                 if (status.equalsIgnoreCase(QueryProgress.COMPLETE.toString())) {
+                  if (isExpired(answer)) {
+                    responseBuilder =
+                        new ResponseBuilder("failed")
+                            .setTypeAndTitle(410, URL_EXPIRED_URN.getUrn())
+                            .setMessage(
+                                "The requested url/resource is expired and no longer available");
+                    handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
+                    return;
+                  }
+                  if (!isUsageWithinLimits(authInfo, answer.getLong("size"))) {
```

Comment:
In this scenario, if a user has a search ID that has already been audited, and they have a usage limit of 500 MB which has been fully consumed for that search ID, what happens if they attempt to access the resource again? Since their remaining limit is now zero, should we allow them to access the resource that has already been audited, or should access be denied? @Gokul-KG @swaminathanvasanth